muskathlon/models/survey_user_input.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, api
+
+
+class SurveyUserInput(models.Model):
+    _inherit = 'survey.user_input'
+
+    @api.multi
+    def write(self, vals):
+        """
+        Automatically complete Medical Survey task when user filled it
+        """
+        res = super(SurveyUserInput, self).write(vals)
+        if vals.get('state') == 'done':
+
+            # Search for medical surveys
+            registrations = self.env['event.registration'].search([
+                ('medical_survey_id', 'in', self.ids),
```

Comment:
I'm sorry but after thinking of it, this won't work because the Muskathlon medical survey is not linked to the registration object. So you should link it like it was done for the regular event and survey.

In object event_registration, you should override method `next_stage` and there search for the survey and link it. We can discuss it together if you need more explanations how to do it.